core/trino-main/src/main/java/io/trino/sql/analyzer/AggregationAnalyzer.java

```diff
@@ -79,6 +79,7 @@
 
```

Comment:
That is what JFR profile tells me. Unfortunately I cannot share the exact query here.
But positions 2-5 in this profile:
![obraz](https://user-images.githubusercontent.com/59822059/201061167-09ae8b61-901e-4670-8b7c-7cff07f241c2.png)
is associated with constructing this map

Reply:
The issue is we use `NodeRef` in a lot of places. Why this particular place is special?

Reply:
> Why this particular place is special?

It is special for my case as it seems to be  performance bottleneck.
> The issue is we use NodeRef in a lot of places

The usage in this particular class in encapsulated, so IMHO this will not cause any issues.

The question is why we were using `NodeRef` so extensively in the first place (I found 681 usages). Maybe I am simply missing something and it does indeed bring some benefits.

Reply:
cc @findepi